src/Network/GRPC/MQTT/Proto.hs

```diff
@@ -0,0 +1,672 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE TemplateHaskellQuotes #-}
+
+-- | This module defines functions used by @grpc-mqtt@ internals for
+--
+-- * Handling IO for protocol buffers.
+--
+-- * Manipulating the protobuf syntax trees.
+--
+-- @since 0.1.0.0
+module Network.GRPC.MQTT.Proto
```

Comment:
Many of these functions could usefully be moved to `proto3-suite`. I don't suggest doing that in this PR though.